### ISSUE TITLE
Ctrl/Shitf Click on internal application link doesn't open in a new browser tab/window

### ISSUE THREAD

**Issue by Andrzej-W** [aspnet/Blazor#867]
Standard behaviour for Ctrl/Shift Click in Firefox/Chrome/Edge is to open link in a new browser tab or window. Unfortunately this does not work for internal links in Blazor application. Middle mouse click doesn't work either (it works like standard left click). Blazor also doesn't respect `target="_blank"` attribute for internal links.

I can right-click on internal link and select "Open in a new tab/window" from context menu - this does work.

**Why we need this behaviour?**
First it is needed for consistency. Second, in LOB applications people often have to see related data from different modules. To do this in the browser they can Ctrl Click on the link to open it in a new browser tab and then drag this tab to the second monitor - it is very convenient.

**Comment by Suchiman** [comment]
On https://blazor-demo.github.io/ with Edge and `CTRL + Click` and `CTRL + Shift + Click` this does work for me. E.g. clicking on Counter opens a new page on https://blazor-demo.github.io/Counter

**Comment by Andrzej-W** [comment]
You are right - in Edge Ctrl+Click works - a new browser **tab** is opened. Shift+Click doesn't work - link should be opened in a new browser **window**. Ctrl+Click and Shift+Click doesn't work as expected in Firefox and Chrome.

**Comment by Suchiman** [comment]
Strange indeed, CTRL / Shift + Click does nothing on Firefox but middle mouse click does work for opening a new tab. Interestingly the only difference Shift makes for me, is that the new tab gets focus immediately vs. opening in the background.

**Comment by Andrzej-W** [comment]
https://blazor-demo.github.io/ is based on Blazor 0.2.0. Middle click works here in all browsers. I have Blazor 0.3.0 on my computer and here middle click works in Edge and Chrome, but **doesn't work in Firefox**.
Firefox 60.0.1
Chrome 66.0.3359.181
Edge Microsoft EdgeHTML 16.16299, Microsoft Edge 41.16299.402.0

**Comment by SteveSandersonMS** [comment]
Thanks for reporting it.

The fix should be a relatively simple tweak in `UriHelper.ts` around line 25 to check whether any `metaKey` flag or similar is present on the `event` and if so, skip the internal-link handling logic.

If anyone wants to submit a PR for it we'd be happy to take it!

**Comment by RemiBou** [comment]
what's your expectation regarding tests for this PR ? E2E ?
